Skate_ERP_preprocessing_IO.m

```diff
@@ -75,326 +66,77 @@
-            EEG.data(x,:) = (EEG.data(x,:)-((EEG.data(EEG.nbchan-2,:))*.5));
-        end
-        
-        %Filter the data with low pass of 30
-               EEG = pop_eegfilt( EEG, .1, 0, [], 0);  %high pass filter
-               EEG = pop_eegfilt( EEG, 0, 30, [], 0);  %low pass filter
-        
-        
-        all_events = length(EEG.event)
-        for i_event = 2:all_events
-            EEG.event(i_event).type = num2str(str2num((EEG.event(i_event).type(2:end))));
-        end
-        
-        %epoch 
-               
-        EEG = pop_epoch( EEG, {  '3'  '5'  }, [-.2  1], 'newname',  sprintf('%s epochs' , setname), 'epochinfo', 'yes'); %Changed from [-.2 1] to [-1 2]. DR
-        EEG = pop_rmbase( EEG, [-200    0]);
-        
-        %         eeglab redraw
-             
-        
-        %    Artifact rejection, trials with range >500 uV
-        EEG = pop_eegthresh(EEG,1,[1:size(EEG.data,1)],-500,500,EEG.xmin,EEG.xmax,0,1);
-        
-        %   EMCP occular correction
-        temp_ocular = EEG.data(end-1:end,:,:); %to save the EYE data for after
-        selection_cards = {'3','5' }; %different bin names, each condition should be separate
-        EEG = gratton_emcp(EEG,selection_cards,{'VEOG'},{'HEOG'}); %this assumes the eye channels are called this
-        EEG.emcp.table %this prints out the regression coefficients
-        EEG.data(end-1:end,:,:) = temp_ocular; %replace the eye data
-        %    Artifact rejection, trials with range >250 uV
-        EEG = pop_rmbase( EEG, [-200 0]); %baseline again since this changed it
-        EEG = pop_eegthresh(EEG,1,[1:size(EEG.data,1)-2],-200,200,EEG.xmin,EEG.xmax,0,1);
-        
-        if i_cond ==1
-        tempEEG1 =   EEG;
-        elseif i_cond==2
-        tempEEG2 = EEG;
-        
-        tempEEG = pop_mergeset(tempEEG1,tempEEG2,1);
-        
-        
-        %now select the corrected trials
-        EEG = pop_selectevent( tempEEG, 'type',5,'renametype','Target','deleteevents','on','deleteepochs','on','invertepochs','off');
-        EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' conds{i_cond} '_Corrected_Target']);
-        EEG = pop_saveset( EEG, 'filename',[subs{i_sub} '_' exp '_facing_out_Corrected_Target.set'],'filepath',[Pathname 'segmentsIO\']);
-        
-        
-        EEG = pop_selectevent( tempEEG, 'type',3 ,'renametype','Standard','deleteevents','on','deleteepochs','on','invertepochs','off');
-        EEG = pop_editset(EEG, 'setname',[subs{i_sub} '_' exp '_' conds{i_cond} '_Corrected_Standard']);
-        EEG = pop_saveset( EEG, 'filename',[subs{i_sub} '_' exp '_facing_out_Corrected_Standard.set'],'filepath',[Pathname 'segmentsIO\']);
         end
         
-    end 
-end 
-commandhistory
-%%
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-%                    GOOFY IN
-%
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-%%
-clear all
-close all
-ccc
+        if is_goofy(i_sub) 
```

Comment:
Why is this code repeated again down here?

Reply:
I'm not sure why you repeat the if elseif end statement twice, that just makes the name and you shouldn't have to run it for both targets and standards

Reply:
This part was confusing. 

The loop to change the names of the variables was identical to the loop we have on the behavioural file, so I can easily see how it's correctly assigning the conditions to the New_conds variable. 

But this is the ERP preprocessing, doesn't the loop need to be ran again to rename the conditions for the targets?

Reply:
no since you can rename both the targets and standards inside the same loop